### PR TITLE
feat: Add product functionality - includes tests, controllers, routes…

### PR DESCRIPTION
**Pull request by Mugisha146**
…, models, services, types, and validations

#### What does this PR do?

Seller can create product,retrieve all product and update product( stock,availability,expireDate ).

#### Describe your changes

Added product model (attribute: id,name,description,price,stock,available,expireDate), product validation, functionality and tests for the endpoint as well as the Swagger Documentation

#### PR Type
What kind of change does this PR introduce?
```
[ ] Feature
[ ] Tests
```

#### How should this be manually tested?

Navigate to the branch and Test the functionality either in Swagger UI or Postman

#### What are the relevant pivotal tracker stories?
[Pivotal Tracker Link](https://www.pivotaltracker.com/story/show/187570741)

### CODE REVIEW

**Review by SHYAKA-Aime — CHANGES_REQUESTED** [review]
Just Try to improve your test coverage

**Review by mutsinziisaac — CHANGES_REQUESTED** [review]
work on the test coverage

File: __test__/product.test.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    sequelizeInstance = await db();
+    await sequelizeInstance.query(
+      "TRUNCATE TABLE products RESTART IDENTITY CASCADE;"
+    );
+    await sequelizeInstance.query(
+      "TRUNCATE TABLE users RESTART IDENTITY CASCADE;"
+    );
+
+    // Create a test admin user and get a token
+    const sellerUser = {
```
Comment (mugishaj092) [inline __test__/product.test.ts:21]: Create those variable in file called variable.utils.ts and then use it here

File: src/database/migrations/20240512215421-create-users.js [hunk 2]
```diff
@@ -1,69 +1,115 @@
...
+
+    // Create 'products' table
+    await queryInterface.createTable("products", {
+      id: {
+        type: Sequelize.UUID,
+        allowNull: false,
+        primaryKey: true,
+        defaultValue: Sequelize.UUIDV4,
+      },
+      name: {
```
Comment (mugishaj092) [inline src/database/migrations/20240512215421-create-users.js:74]: create new product migration file

File: src/database/models/product.model.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import {
+  ProductAttributes,
+  ProductCreationAttributes,
+} from "../../types/product.types";
+
+class Product
+  extends Model<ProductAttributes, ProductCreationAttributes>
+  implements ProductAttributes
+{
+  declare id: string;
```
Comment (mugishaj092) [inline src/database/models/product.model.ts:12]: You forget to add field of picture

**Review by mugishaj092 — CHANGES_REQUESTED** [review]
Good job so far but i give few feedbacks check comments

**Pull request closed without merging**